viewer/packages/360-images/src/cache/Image360LoadingCache.ts

```diff
@@ -30,25 +36,46 @@ export class Image360LoadingCache {
 
     const inflightEntity = this._inFlightEntities.get(entity);
     if (inflightEntity !== undefined) {
-      return inflightEntity;
+      return inflightEntity.load360Image;
     }
 
-    const load360Image = entity.load360Image();
-    this._inFlightEntities.set(entity, load360Image);
-
-    await load360Image;
-
-    if (this._loaded360Images.length === this._cacheSize) {
-      this.purgeLastRecentlyUsedInvisibleEntity();
+    if (this._inFlightEntities.size === this._inFlightCacheSize) {
+      this.abortLastRecentlyReqestedEntity();
     }
 
-    this._loaded360Images.unshift(entity);
-    this._inFlightEntities.delete(entity);
+    const { signal, abort } = this.createAbortSignal();
+    const load360Image = entity
+      .load360Image(signal)
+      .catch(e => {
+        if (signal.aborted || e === 'Aborted') {
+          Log.info('Abort warning: ' + e);
+        } else {
+          Log.warn('Failed to load 360 image: ' + e);
+        }
+        return Promise.reject();
+      })
+      .then(
+        () => {
+          if (this._loaded360Images.length === this._imageCacheSize) {
+            this.purgeLastRecentlyUsedInvisibleEntity();
+          }
+          this._loaded360Images.unshift(entity);
+        },
+        () => {
+          return Promise.resolve();
+        }
```

Comment:
Is this necessary? What does it achieve?

Reply:
I tested and I do not need the `return Promise.resolve()`. But I do need the "on reject" function to stay so that the function as a whole does not throw.